ReactiveCocoa/Swift/Property.swift

```diff
@@ -299,3 +313,33 @@ public func <~ <P: MutablePropertyType>(property: P, producer: SignalProducer<P.
 public func <~ <Destination: MutablePropertyType, Source: PropertyType where Source.Value == Destination.Value>(destinationProperty: Destination, sourceProperty: Source) -> Disposable {
 	return destinationProperty <~ sourceProperty.producer
 }
+
+
+/// Type erased wrapper for composable PropertyType.
+private class _AnyPropertyBoxBase<Value>: PropertyType {
+	var producer: SignalProducer<Value, NoError> {
+		fatalError()
```

Comment:
The type erased wrapper is to workaround the lack of way to legally define a non-escaping and `rethrows` stored closure in Swift 2's type system, as far as I tried. This prohibits the `withValue` method from being wrapped in a closure. As a matter of fact, this pattern is used in the Swift STL to workaround Swift 2's lacking existential. (https://github.com/apple/swift/blob/master/stdlib/public/core/ExistentialCollection.swift.gyb).

There are alternatives that can be considered though, e.g.
1. an escaping, non-throwing version of `withValue` and `modify`, specifically for property composition
2. a private `AnyProperty` initialiser that assumes a buffering producer.
2. wait for Swift 3.0 which hopefully comes with partially/fully conformed existentials. ;-)